### DISCUSSION TITLE
composition + vuex?

### DISCUSSION THREAD

**Discussion by Esurnir** [intlify/vue-i18n#535] [Q&A]
Hello, sorry if this is a stupid question but : If I turn legacy: off. is it possible for me to use i18n within the vuex store ? (I used vuei18n within a composable mixed with the store, but i was curious if i could use it straight from the store).

**Comment by kazupon** [comment]
Hello!

I have not used the Composition API style vue-i18n API in Vuex, so I can't vouch for whether it works or not.
However, as a vue-i18n author who knows about inner part implementation, `useI18n` only works with the `setup` inner part because it calls `getCurrentInstance` in the inner part and depends on ComponentPublicInstance.

The `global` property of the i18n instance retrieved by the `createI18n` function is the composer instance returned by `useI18n` (the instance that provides the API for `t`, `d`, etc.), so I think it can be used by Vuex.

By the way, I don't understand why you are using vue-i18n with Vuex in your use case.
Vuex is a state management layer. vue-i18n, on the other hand, is a library about the View layer, dealing with messages, dates, times, and numbers.
Therefore, I think it is bad practice for each module to be used in a way that depends on each other since each module is given its own debt as a library.

**Reply by ataldev** [reply]
Sometimes the translations are not predefined in JSON files and must be loaded using an API. I have seen lots of websites where the admin is able to add languages, and then add translations to that language.
